bot/cogs/translate_cog.py

```diff
@@ -0,0 +1,179 @@
+import json
+import logging
+import discord
+import aiohttp
+import uuid
+import bot.extensions as ext
+import discord.ext.commands as commands
+from discord.ext.commands.errors import UserInputError
+from bot.bot_secrets import BotSecrets
+from bot.consts import Colors
+
+log = logging.getLogger(__name__)
+
+language_name_to_short_code = {
+    "afrikaans": "af",
+    "arabic": "ar",
+    "bulgarian": "bg",
+    "catalan": "ca",
+    "chinese simplified": "zh-Hans",
+    "chinese traditional":	"zh-Hant",
+    "croatian": "hr",
+    "czech": "cs",
+    "danish": "da",
+    "dutch": "nl",
+    "english": "en",
+    "estonian": "et",
+    "finnish": "fi",
+    "french": "fr",
+    "german": "de",
+    "greek": "el",
+    "gujarati": "gu",
+    "haitian creole": "ht",
+    "hebrew": "he",
+    "hindi": "hi",
+    "hungarian":"hu",
+    "icelandic": "is",
+    "indonesian": "id",
+    "irish": "ga",
+    "italian": "it",
+    "japanese": "ja",
+    "klingon": "tlh-Latn",
+    "korean": "ko",
+    "kurdish (central)": "ku-Arab",
+    "latvian": "lv",
+    "lithuanian": "lt",
+    "malay": "ms",
+    "maltese": "mt",
+    "norwegian": "nb",
+    "pashto": "ps",
+    "persian": "fa",
+    "polish": "pl",
+    "portuguese": "pt",
+    "romanian": "ro",
+    "russian": "ru",
+    "serbian (cyrillic)": "sr-Cyrl",
+    "serbian (latin)": "sr-Latn",
+    "slovak": "sk",
+    "slovenian": "sl",
+    "spanish": "es",
+    "swahili": "sw",
+    "swedish": "sv",
+    "tahitian": "ty",
+    "thai": "th",
+    "turkish": "tr",
+    "ukrainian": "uk",
+    "urdu": "ur",
+    "vietnamese": "vi",
+    "welsh": "cy",
+    "yucatec maya": "yua"
+}
+
+language_short_code_to_name = {value : key for (key, value) in language_name_to_short_code.items()}
+
+headers = {
+    'Ocp-Apim-Subscription-Key': BotSecrets.get_instance().translator_subscription_key,
+    'Ocp-Apim-Subscription-Region': 'global',
+    'Content-type': 'application/json',
+    'X-ClientTraceId': str(uuid.uuid4())
+}
+
+
+TRANSLATE_API_URL = "https://api.cognitive.microsofttranslator.com/translate"
+
+class TranslateCog(commands.Cog):
+
+    def __init__(self, bot):
+        self.bot = bot
+        self._last_member = None
+
+    @ext.command()
+    @ext.long_help('Allows you to translate words or sentences by either specifying both the input and output language with the text to translate, or just the output language and the text to translate')
+    @ext.short_help('Translates words or phrases between two languages')
+    @ext.example(('translate en spanish Hello', 'translate german Hello'))
+    async def translate(self, ctx, *input: str):
+        if len(input) < 2:
+            raise UserInputError("Incorrect Number of Arguments. Minimum of 2 arguments")
```

Comment:
we should send a list of valid languages here in stead of raising an exception

Reply:
Wait. send a list of valid languages here or at the error that says its not a valid language? Right here it's just saying that there are not enough arguments to do anything